src/vcpkg/commands.new.cpp

```diff
@@ -1,15 +1,217 @@
 #include <vcpkg/base/basic_checks.h>
 
 #include <vcpkg/commands.new.h>
-#include <vcpkg/configure-environment.h>
+#include <vcpkg/configuration.h>
+#include <vcpkg/registries.h>
 #include <vcpkg/tools.h>
 #include <vcpkg/vcpkgcmdarguments.h>
+#include <vcpkg/vcpkgpaths.h>
+#include <vcpkg/versions.h>
+
+namespace
+{
+    using namespace vcpkg;
+
+    static constexpr StringLiteral OPTION_APPLICATION = "application";
+    static constexpr StringLiteral OPTION_SINGLE_FILE = "single-file";
+
+    static constexpr StringLiteral SETTING_NAME = "name";
+    static constexpr StringLiteral SETTING_VERSION = "version";
+
+    static constexpr StringLiteral OPTION_VERSION_RELAXED = "version-relaxed";
+    static constexpr StringLiteral OPTION_VERSION_DATE = "version-date";
+    static constexpr StringLiteral OPTION_VERSION_STRING = "version-string";
+
+    const CommandSwitch SWITCHES[] = {
+        {OPTION_APPLICATION, "Create an application manifest (don't require name or version)."},
+        {OPTION_SINGLE_FILE, "Embed vcpkg-configuration.json into vcpkg.json."},
+        {OPTION_VERSION_RELAXED,
+         "Interpret --version as a relaxed-numeric version. (Nonnegative numbers separated by dots)"},
+        {OPTION_VERSION_DATE, "Interpret --version as an ISO 8601 date. (YYYY-MM-DD)"},
+        {OPTION_VERSION_STRING, "Interpret --version as a string with no ordering behavior."},
+    };
+
+    const CommandSetting SETTINGS[] = {
+        {SETTING_NAME, "Name for the new manifest."},
+        {SETTING_VERSION, "Version for the new manifest."},
+    };
+
+    const CommandStructure COMMAND_STRUCTURE = {
+        create_example_string(R"###(new --name=example --version=1.0 --version-kind=relaxed)###"),
+        0,
+        0,
+        {SWITCHES, SETTINGS, {}},
+        nullptr,
+    };
+}
 
 namespace vcpkg::Commands
 {
+    ExpectedL<Json::Object> build_prototype_manifest(const std::string* name,
+                                                     const std::string* version,
+                                                     bool option_application,
+                                                     bool option_version_relaxed,
+                                                     bool option_version_date,
+                                                     bool option_version_string)
+    {
```

Comment:
This is just personal preference, but the function could be easier to digest if it were split into smaller chunks for each option and just pass the manifest as an argument.

Reply:
I'm not sure I understand: the manifest is the return value? Can you show the declarations you'd be looking for?